atlasdb-timelock-server/src/main/java/com/palantir/atlasdb/timelock/atomix/AtomixRetryer.java

```diff
@@ -0,0 +1,55 @@
+/**
+ * Copyright 2016 Palantir Technologies
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.palantir.atlasdb.timelock.atomix;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.function.Supplier;
+
+import com.github.rholder.retry.RetryException;
+import com.github.rholder.retry.Retryer;
+import com.github.rholder.retry.RetryerBuilder;
+import com.github.rholder.retry.StopStrategies;
+import com.google.common.base.Throwables;
+import com.google.common.util.concurrent.Futures;
+import com.google.common.util.concurrent.UncheckedExecutionException;
+
+import io.atomix.copycat.session.ClosedSessionException;
+
+public final class AtomixRetryer {
+    public static final int RETRY_ATTEMPTS = 3;
+
+    private static final Retryer<Object> RETRYER = RetryerBuilder.newBuilder()
+            .retryIfException(e -> e instanceof UncheckedExecutionException
+                    && e.getCause() instanceof ClosedSessionException)
```

Comment:
Do we want to log or have some indication that we're retrying as a result of errors for observability and overall system health?